src/runtime/pkg/resourcecontrol/cgroups.go

```diff
@@ -148,10 +145,13 @@ func NewResourceController(path string, resources *specs.LinuxResources) (Resour
 		if err != nil {
 			return nil, err
 		}
-		cgroup, err = cgroupsv2.NewManager(unifiedMountpoint, cgroupPath, cgroupsv2.ToResources(resources))
+		cgroup, err = cgroupsv2.NewManager(UnifiedMountpoint, cgroupPath, cgroupsv2.ToResources(resources))
 		if err != nil {
 			return nil, err
 		}
+		if !sandboxCgroupOnly {
+			SetThreadedMode(cgroupPath)
```

Comment:
Does it break systemd cgroup manager?

Reply:
In the case of systemd managed cgroups, we'd get something like `cgroupPath = /slice:scopeprefix:name`. It doesn't seem that `SetThreadedMode` is capable of handling such kind of path.

Reply:
> In the case of systemd managed cgroups, we'd get something like `cgroupPath = /slice:scopeprefix:name`. It doesn't seem that `SetThreadedMode` is capable of handling such kind of path.

When `cgroup driver = systemd` and `sandbox_cgroup_only = false`, the following cgroup hierarchy will be generated.

```
root@k8s-worker01:~# ls /sys/fs/cgroup/kubepods-besteffort-podcb76d0a3_83c7_4ad2_922c_095eeea7e6fe.slice\:cri-containerd\:6595af1eab666f43d9bd8ca4e12347a40e3a2bac81393dc9fa3318e15050ece0/
cgroup.controllers      cgroup.stat             cpuset.cpus            cpu.uclamp.max   io.stat              memory.low        overhead      sandbox
cgroup.events           cgroup.subtree_control  cpuset.cpus.effective  cpu.uclamp.min   io.weight            memory.max        pids.current
cgroup.freeze           cgroup.threads          cpuset.cpus.partition  cpu.weight       memory.current       memory.min        pids.events
cgroup.max.depth        cgroup.type             cpuset.mems            cpu.weight.nice  memory.events        memory.oom.group  pids.max
cgroup.max.descendants  cpu.max                 cpuset.mems.effective  io.max           memory.events.local  memory.pressure   rdma.current
cgroup.procs            cpu.pressure            cpu.stat               io.pressure      memory.high          memory.stat       rdma.max
```

To be honest, this is not a reasonable way to manage cgroups, because cgroups generated using cgroupfs cannot be managed with systemd. And it does not follow the cgroup hierarchy of kubernetes, although in fact the source of statistical metrics is the cgroup in the agent, and the host cgroup does not affect the metrics.

If this approach is acceptable, I consider could using the `kata` as  parent directory to all cgroups created in the form of `cgroupPath = /slice:scopeprefix:name`, for example: `cgroupPath = /kata/slice:scopeprefix:name/sandbox` and `overheadPath = /kata/slice:scopeprefix:name/overhead`.

If this approach is unacceptable, it may be possible need to report an error when the user configures `cgroup driver = systemd` and `sandbox_cgroup_only = false` telling the user: Kata does not support using systemd to manage cgroups when `sandbox_cgroup_only = false`.